crates/ruff_python_formatter/README.md

```diff
@@ -130,8 +141,67 @@ impl FormatNodeRule<ExprList> for FormatExprList {
 }
 ```
 
-Comments are categorized into `Leading`, `Trailing` and `Dangling`, you can override this in
-`place_comment`.
+A common challenge is that we want to attach comments to tokens (think keywords and syntactically
+meaningful characters such as `:`) that have no node on their own. A slightly simplified version of
+the `while` node in our AST looks like the following:
+
+```rust
+pub struct StmtWhile {
+    pub range: TextRange,
+    pub test: Box<Expr<TextRange>>,
+    pub body: Vec<Stmt<TextRange>>,
+    pub orelse: Vec<Stmt<TextRange>>,
+}
+```
+
+That means in
+
+```python
+while True:  # Trailing cond comment
+    if f():
+        break
+    # trailing while comment
+# leading else comment
+else:
+    print("while-else")
+```
+
+the `else` has no node, we're just getting the statements in its body.
+
+By default, the comment would get misattributed and moved to the `break` (the first node before the
```

Comment:
(mis)attributed: maybe i'm missing something but to me, talking about "attribution" sounds like the work of the parser whereas we are talking about format output, right?